File: SelectionSort/js/SelectionSort.js
'use strict';

class SelectionSort {
	findIndexOfLowest(numberList) {
		var lowest = 0;
		numberList.forEach((val, i) => {
            lowest = val < numberList[lowest] ? i : lowest;
        });
		return lowest;
	}

    swapFromTo(numberList, from, to) {
        let tempFrom = numberList[from];
        numberList[from] = numberList[to];
        numberList[to] = tempFrom;
        return numberList;
    }

    sort(numberList) {
        numberList.forEach( (val,i) => {
            let indexOfLowest = i + this.findIndexOfLowest(numberList.slice(i));
            numberList = this.swapFromTo(numberList, i, indexOfLowest);
        });
        return numberList;
    }
}

File: SelectionSort/py/SelectionSort.py
#!/usr/local/bin/python3

class SelectionSort:
    def  __init__(self):
        pass

    def findIndexOfLowestInList(self, numberList):
        lowest = 0

        for i,num in enumerate(numberList):
            if num < numberList[lowest]:
                lowest = i
        return lowest

    def swapFromTo(self,numberList,fromPos,toPos):
        tempNum = numberList[fromPos]
        numberList[fromPos] = numberList[toPos]
        numberList[toPos] = tempNum
        return numberList

    def sort(self,numberList):
        for i,num in enumerate(numberList):
            numberList = self.swapFromTo(numberList, i, i+self.findIndexOfLowestInList(numberList[i:]))
        return numberList
